src/Authenticator/SessionAuthenticator.php

```diff
@@ -84,6 +84,7 @@ public function authenticate(ServerRequestInterface $request, ResponseInterface
     public function persistIdentity(ServerRequestInterface $request, ResponseInterface $response, $identity)
     {
         $sessionKey = $this->getConfig('sessionKey');
+        $request->getAttribute('session')->renew();
```

Comment:
This might get called on every request when this is merged up to 2.x. This block https://github.com/cakephp/authentication/blob/deacd5a03378e8284e873f17cd932cf6eae2876c/src/Middleware/AuthenticationMiddleware.php#L130-L133 runs each time the user has been authenticated which happens on every request with session authenticator.

Reply:
I guess the session authenticator will have to first check if a persisted identity already exists and renew session id only if no identity is currently exists in session or existing one is not the same as new one.